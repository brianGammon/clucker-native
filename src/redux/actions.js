import firebase from 'react-native-firebase';
import { metaTypes, actionTypes as a } from './constants';

export function firebaseListenRequested(ref, metaType) {
  return {
    type: a.LISTEN_REQUESTED,
    payload: { ref },
    meta: { type: metaType },
  };
}

export function firebaseListenRejected(error, metaType) {
  return {
    type: a.LISTEN_REJECTED,
    payload: { error },
    meta: { type: metaType },
  };
}

export function firebaseListenFulfilled({ key, data }, metaType) {
  return {
    type: a.LISTEN_FULFILLED,
    payload: { key, data },
    meta: { type: metaType },
  };
}

export function firebaseListenChildAdded(key, data, metaType) {
  return {
    type: a.LISTEN_CHILD_ADDED,
    payload: { key, data },
    meta: { type: metaType },
  };
}

export function firebaseListenChildChanged(key, data, metaType) {
  return {
    type: a.LISTEN_CHILD_CHANGED,
    payload: { key, data },
    meta: { type: metaType },
  };
}

export function firebaseListenChildRemoved(key, metaType) {
  return {
    type: a.LISTEN_CHILD_REMOVED,
    payload: { key },
    meta: { type: metaType },
  };
}

export function firebaseUpdateRequested(payload, metaType) {
  return {
    type: a.UPDATE_REQUESTED,
    payload,
    meta: { type: metaType },
  };
}

export function firebaseUpdateRejected(error, metaType) {
  return {
    type: a.UPDATE_REJECTED,
    payload: { error },
    meta: { type: metaType },
  };
}

export function firebaseUpdateFulfilled(metaType) {
  return {
    type: a.UPDATE_FULFILLED,
    payload: {},
    meta: { type: metaType },
  };
}

export function firebaseRemoveRequested(payload, metaType) {
  return {
    type: a.REMOVE_REQUESTED,
    payload,
    meta: { type: metaType },
  };
}

export function firebaseRemoveRejected(error, metaType) {
  return {
    type: a.REMOVE_REJECTED,
    payload: { error },
    meta: { type: metaType },
  };
}

export function firebaseRemoveFulfilled(metaType) {
  return {
    type: a.REMOVE_FULFILLED,
    payload: {},
    meta: { type: metaType },
  };
}

export function firebaseListenRemoved(clearData, metaType) {
  return {
    type: a.LISTEN_REMOVED,
    payload: { clearData },
    meta: { type: metaType },
  };
}

export function firebaseRemoveListenerRequested(clearData, metaType) {
  return {
    type: a.REMOVE_LISTENER_REQUESTED,
    payload: { clearData },
    meta: { type: metaType },
  };
}

export function firebaseRemoveAllListenersRequested() {
  return {
    type: a.REMOVE_ALL_LISTENERS_REQUESTED,
    payload: { clearData: true },
  };
}

export function listenToChickens(flockId) {
  const ref = firebase.database().ref(`chickens/${flockId}`);
  return firebaseListenRequested(ref, metaTypes.chickens);
}

export function listenToEggs(flockId) {
  const ref = firebase.database().ref(`eggs/${flockId}`);
  return firebaseListenRequested(ref, metaTypes.eggs);
}

export function listenToUserSettings(uid) {
  const ref = firebase.database().ref(`userSettings/${uid}`);
  return firebaseListenRequested(ref, metaTypes.userSettings);
}

export function removeUserSettingsListenerRequested() {
  return firebaseRemoveListenerRequested(false, metaTypes.userSettings);
}

// TODO: remove this, no need
export function updateUserSettingsRequested(
  uid,
  displayName,
  currentFlockId,
  flocks,
) {
  return firebaseUpdateRequested(
    {
      uid,
      displayName,
      currentFlockId,
      flocks,
    },
    metaTypes.userSettings,
  );
}

export function removeUserSettingsRequested(uid) {
  return firebaseRemoveRequested({ uid }, metaTypes.userSettings);
}

// TODO: remove refs from actions module
export function getFlock(flockId) {
  const ref = firebase.database().ref(`flocks/${flockId}`);
  return {
    type: a.GET_FLOCK_REQUESTED,
    payload: { ref },
    meta: { type: metaTypes.flocks },
  };
}

export function getFlockRejected(error) {
  return {
    type: a.GET_FLOCK_REJECTED,
    payload: { error },
    meta: { type: metaTypes.flocks },
  };
}

export function getFlockFulfilled(flock) {
  return {
    type: a.GET_FLOCK_FULFILLED,
    payload: flock,
    meta: { type: metaTypes.flocks },
  };
}

export function setInitialUrl(url) {
  return {
    type: a.SET_INITIAL_URL,
    payload: url,
  };
}

export function removeInitialUrl() {
  return {
    type: a.REMOVE_INITIAL_URL,
  };
}
