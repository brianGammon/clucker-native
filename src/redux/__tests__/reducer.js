import * as actions from '../actions';
import {
  actionTypes, metaTypes, appStates, authTypes,
} from '../constants';
import firebaseReducer from '../reducer';

describe('firebaseReducer reducer', () => {
  let sampleState;
  beforeEach(() => {
    sampleState = {
      appState: appStates.STARTING,
      initialUrl: null,
      auth: {
        inProgress: false,
        errors: {
          signIn: null,
          signUp: null,
          resetPassword: null,
        },
        user: null,
      },
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: false,
        error: null,
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
  });

  test(actionTypes.AUTH_STATUS_LOGGED_IN, () => {
    const user = { uid: 'testId', email: 'test@example.com' };
    const action = actions.authStatusChanged(user);
    expect(firebaseReducer(sampleState, action)).toMatchSnapshot();
  });

  test(actionTypes.AUTH_STATUS_LOGGED_OUT, () => {
    const user = { uid: 'testId', email: 'test@example.com' };
    const { errors } = sampleState.auth;
    const initialState = {
      appState: appStates.STARTING,
      auth: {
        inProgress: false,
        errors,
        user,
      },
    };
    const action = actions.authStatusChanged(null);
    const expectedState = {
      appState: appStates.READY,
      auth: {
        inProgress: false,
        errors,
        user: null,
      },
    };
    expect(firebaseReducer(initialState, action)).toEqual(expectedState);
  });

  test(actionTypes.AUTH_ACTION_REQUESTED, () => {
    const action = {
      type: actionTypes.AUTH_ACTION_REQUESTED,
      payload: { email: 'test@example.com', password: 'test123' },
      meta: { type: authTypes.signIn },
    };
    const expectedState = {
      ...sampleState,
      auth: {
        ...sampleState.auth,
        inProgress: true,
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(`${actionTypes.AUTH_ACTION_REQUESTED} after an error`, () => {
    const {
      auth: { errors },
    } = sampleState;
    sampleState.auth = {
      inProgress: false,
      errors: { ...errors, signIn: 'Error from before' },
      user: null,
    };

    const action = {
      type: actionTypes.AUTH_ACTION_REQUESTED,
      payload: { email: 'test@example.com', password: 'test123' },
      meta: { type: authTypes.signIn },
    };
    const expectedState = {
      ...sampleState,
      auth: {
        ...sampleState.auth,
        inProgress: true,
        errors,
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.AUTH_ACTION_FULFILLED, () => {
    const { auth: { errors } } = sampleState;
    sampleState.auth = { inProgress: true, errors, user: { uid: '123' } };
    const action = { type: actionTypes.AUTH_ACTION_FULFILLED, meta: { type: authTypes.signIn } };
    const expectedState = {
      ...sampleState,
      auth: {
        inProgress: false,
        errors,
        user: { uid: '123' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.AUTH_ACTION_REJECTED, () => {
    const { auth: { errors } } = sampleState;
    Object.keys(authTypes).forEach((authType) => {
      sampleState.auth = { inProgress: true, errors, user: null };
      const error = new Error('Some error occurred');
      const action = { type: actionTypes.AUTH_ACTION_REJECTED, payload: { error }, meta: { type: authType } };
      const expectedState = {
        ...sampleState,
        auth: {
          inProgress: false,
          errors: { ...errors, [action.meta.type]: error.message },
          user: null,
        },
      };
      expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
    });
  });

  test(actionTypes.SET_INITIAL_URL, () => {
    const initialState = {
      someOtherState: 1,
      initialUrl: null,
    };
    const action = actions.setInitialUrl('someUrl');
    const expectedState = {
      someOtherState: 1,
      initialUrl: 'someUrl',
    };
    expect(firebaseReducer(initialState, action)).toEqual(expectedState);
  });

  test(actionTypes.REMOVE_INITIAL_URL, () => {
    const initialState = {
      someOtherState: 1,
      initialUrl: 'someUrl',
    };
    const action = actions.removeInitialUrl();
    const expectedState = {
      someOtherState: 1,
      initialUrl: null,
    };
    expect(firebaseReducer(initialState, action)).toEqual(expectedState);
  });

  test(actionTypes.CLEAR_ERROR, () => {
    const action = {
      type: actionTypes.CLEAR_ERROR,
      meta: { type: metaTypes.chickens },
    };
    const propertyState = sampleState.chickens;
    const initialState = {
      ...sampleState,
      chickens: {
        ...propertyState,
        error: 'Test error message',
      },
    };
    const expectedState = {
      ...sampleState,
      chickens: { ...propertyState, error: null },
    };
    expect(firebaseReducer(initialState, action)).toEqual(expectedState);
  });

  test(actionTypes.CLEAR_AUTH_ERROR, () => {
    const action = {
      type: actionTypes.CLEAR_AUTH_ERROR,
    };
    const { auth } = sampleState;
    const initialState = {
      ...sampleState,
      auth: {
        ...sampleState.auth,
        errors: { ...auth.errors, signIn: 'Some error occurred' },
      },
    };
    const expectedState = {
      ...sampleState,
      auth,
    };
    expect(firebaseReducer(initialState, action)).toEqual(expectedState);
  });

  test(actionTypes.CREATE_REQUESTED, () => {
    const initialState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: false,
        error: null,
        data: {},
      },
    };
    const payload = {};
    const action = actions.firebaseCreateRequested(
      payload,
      metaTypes.chickens,
    );
    const expectedState = {
      ...initialState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: true,
        error: null,
        data: {},
      },
    };
    expect(firebaseReducer(initialState, action)).toEqual(expectedState);
  });

  test(actionTypes.CREATE_FULFILLED, () => {
    const initialState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: true,
        error: null,
        data: {},
      },
    };
    const action = actions.firebaseCreateFulfilled(metaTypes.chickens);
    const expectedState = {
      ...initialState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: false,
        error: null,
        data: {},
      },
    };
    expect(firebaseReducer(initialState, action)).toEqual(expectedState);
  });

  test(actionTypes.CREATE_REJECTED, () => {
    const initialState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: true,
        error: null,
        data: {},
      },
    };
    const error = new Error('Some error message');
    const action = actions.firebaseCreateRejected(
      error,
      metaTypes.chickens,
    );
    const expectedState = {
      ...initialState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: false,
        error: error.message,
        data: {},
      },
    };
    expect(firebaseReducer(initialState, action)).toEqual(expectedState);
  });

  test(actionTypes.UPDATE_REQUESTED, () => {
    const action = actions.firebaseUpdateRequested(
      { item1: 'updateItem1Value' },
      metaTypes.chickens,
    );
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: true,
        error: null,
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });
  test(actionTypes.UPDATE_REJECTED, () => {
    sampleState[metaTypes.chickens].inProgress = true;

    const error = new Error('test error');
    const action = actions.firebaseUpdateRejected(
      error,
      metaTypes.chickens,
    );

    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: false,
        error: 'test error',
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.UPDATE_FULFILLED, () => {
    sampleState[metaTypes.chickens].inProgress = true;

    const action = actions.firebaseUpdateFulfilled(metaTypes.chickens);
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: false,
        error: null,
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.REMOVE_REQUESTED, () => {
    const action = actions.firebaseRemoveRequested(
      { key: 'item1' },
      metaTypes.chickens,
    );
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: true,
        error: null,
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.REMOVE_REJECTED, () => {
    sampleState[metaTypes.chickens].inProgress = true;

    const error = new Error('test error message');
    const action = actions.firebaseRemoveRejected(
      error,
      metaTypes.chickens,
    );
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: false,
        error: error.message,
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.REMOVE_FULFILLED, () => {
    sampleState[metaTypes.chickens].inProgress = true;

    const action = actions.firebaseRemoveFulfilled(metaTypes.chickens);
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: false,
        error: null,
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.LISTEN_REQUESTED, () => {
    const ref = {};
    const action = actions.firebaseListenRequested(ref, metaTypes.chickens);
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: true,
        error: null,
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.LISTEN_REJECTED, () => {
    sampleState[metaTypes.chickens].inProgress = true;

    const error = new Error('error');
    const action = actions.firebaseListenRejected(
      error,
      metaTypes.chickens,
    );
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: true,
        inProgress: false,
        error: error.message,
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.LISTEN_FULFILLED, () => {
    sampleState[metaTypes.chickens].data = {};

    const data = { 1: { text: 'hello' }, 2: { text: 'world' } };
    const action = actions.firebaseListenFulfilled(
      data,
      metaTypes.chickens,
    );
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: true,
        inProgress: false,
        error: null,
        data,
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.CHILD_ADDED, () => {
    const childKey = 'item3';
    const childData = 'item3value';
    const action = actions.firebaseListenChildAdded(
      childKey,
      childData,
      metaTypes.chickens,
    );
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: false,
        error: null,
        data: { item1: 'item1value', item2: 'item2value', item3: 'item3value' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.CHILD_CHANGED, () => {
    const childKey = 'item2';
    const childData = 'updatedItem2value';
    const action = actions.firebaseListenChildChanged(
      childKey,
      childData,
      metaTypes.chickens,
    );
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: false,
        error: null,
        data: { item1: 'item1value', item2: 'updatedItem2value' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.CHILD_REMOVED, () => {
    const childKey = 'item1';
    const action = actions.firebaseListenChildRemoved(
      childKey,
      metaTypes.chickens,
    );
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: false,
        inProgress: false,
        error: null,
        data: { item2: 'item2value' },
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(`${actionTypes.LISTEN_REMOVED} clear data false`, () => {
    const action = actions.firebaseListenRemoved(false, metaTypes.chickens);
    const state = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: true,
        inProgress: true,
        error: 'Some error from before',
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: true,
        inProgress: true,
        error: null,
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
    expect(firebaseReducer(state, action)).toEqual(expectedState);
  });

  test(`${actionTypes.LISTEN_REMOVED} clear data true`, () => {
    const action = actions.firebaseListenRemoved(true, metaTypes.chickens);
    const state = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: true,
        inProgress: true,
        error: 'Some error from before',
        data: { item1: 'item1value', item2: 'item2value' },
      },
    };
    const expectedState = {
      ...sampleState,
      [metaTypes.chickens]: {
        initialized: true,
        inProgress: true,
        error: null,
        data: {},
      },
    };
    expect(firebaseReducer(state, action)).toEqual(expectedState);
  });

  test(actionTypes.DELETE_CHICKEN_REQUESTED, () => {
    const action = {
      type: actionTypes.DELETE_CHICKEN_REQUESTED,
    };
    const expectedState = {
      ...sampleState,
      deleteChicken: { inProgress: true, error: null },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.DELETE_CHICKEN_FULFILLED, () => {
    const initialState = {
      ...sampleState,
      deleteChicken: {
        inProgress: true,
        error: 'test',
      },
    };
    const action = {
      type: actionTypes.DELETE_CHICKEN_FULFILLED,
    };
    const expectedState = {
      ...sampleState,
      deleteChicken: { inProgress: false, error: null },
    };
    expect(firebaseReducer(initialState, action)).toEqual(expectedState);
  });

  test(actionTypes.DELETE_CHICKEN_REJECTED, () => {
    const error = new Error('Test error message');
    const action = {
      type: actionTypes.DELETE_CHICKEN_REJECTED,
      payload: { error },
    };
    const expectedState = {
      ...sampleState,
      deleteChicken: { inProgress: false, error: error.message },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test(actionTypes.SAVE_CHICKEN_REQUESTED, () => {
    const action = {
      type: actionTypes.SAVE_CHICKEN_REQUESTED,
    };
    const expectedState = {
      ...sampleState,
      chickens: {
        ...sampleState.chickens,
        inProgress: true,
      },
    };
    expect(firebaseReducer(sampleState, action)).toEqual(expectedState);
  });

  test('bogus action does nothing', () => {
    const action = {
      type: 'DO_NOT_TOUCH_STATE_ACTION',
    };
    expect(firebaseReducer(sampleState, action)).toBe(sampleState);
  });

  test('undefined initial state', () => {
    const action = {
      type: 'DO_NOT_TOUCH_STATE_ACTION',
    };
    expect(firebaseReducer(undefined, action)).toMatchSnapshot();
  });

  test('missing initial state', () => {
    expect(firebaseReducer()).toMatchSnapshot();
  });
});
