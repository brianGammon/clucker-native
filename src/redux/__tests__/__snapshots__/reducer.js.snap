// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`firebaseReducer reducer AUTH_STATUS_LOGGED_IN 1`] = `
Object {
  "addForm": Object {
    "error": "",
    "inProgress": false,
  },
  "appState": "READY",
  "auth": Object {
    "error": "",
    "inProgress": false,
    "user": Object {
      "email": "test@example.com",
      "uid": "testId",
    },
  },
  "flocks": Object {
    "data": Object {
      "flock1": "item1value",
      "flock2": "item2value",
    },
    "error": "",
    "inProgress": false,
  },
  "initialUrl": null,
  "joinForm": Object {
    "error": "",
    "inProgress": false,
  },
  "userSettings": Object {
    "data": Object {
      "item1": "item1value",
      "item2": "item2value",
    },
    "error": "",
    "inProgress": false,
  },
}
`;

exports[`firebaseReducer reducer GET_FLOCK_FULFILLED 1`] = `
Object {
  "addForm": Object {
    "error": "",
    "inProgress": false,
  },
  "appState": "STARTING",
  "auth": Object {
    "error": "",
    "inProgress": false,
    "user": null,
  },
  "flocks": Object {
    "data": Object {
      "flock1": Object {
        "name": "Flock 1",
        "ownedBy": "user1",
      },
    },
    "error": "",
    "inProgress": false,
  },
  "initialUrl": null,
  "joinForm": Object {
    "error": "",
    "inProgress": false,
  },
  "userSettings": Object {
    "data": Object {
      "item1": "item1value",
      "item2": "item2value",
    },
    "error": "",
    "inProgress": false,
  },
}
`;

exports[`firebaseReducer reducer GET_FLOCK_FULFILLED 2`] = `
Object {
  "addForm": Object {
    "error": "",
    "inProgress": false,
  },
  "appState": "STARTING",
  "auth": Object {
    "error": "",
    "inProgress": false,
    "user": null,
  },
  "flocks": Object {
    "data": Object {
      "flock1": Object {
        "name": "Flock 1",
        "ownedBy": "user1",
      },
      "flock2": Object {
        "name": "Flock 2",
        "ownedBy": "user2",
      },
    },
    "error": "",
    "inProgress": false,
  },
  "initialUrl": null,
  "joinForm": Object {
    "error": "",
    "inProgress": false,
  },
  "userSettings": Object {
    "data": Object {
      "item1": "item1value",
      "item2": "item2value",
    },
    "error": "",
    "inProgress": false,
  },
}
`;

exports[`firebaseReducer reducer GET_FLOCK_REJECTED 1`] = `
Object {
  "addForm": Object {
    "error": "",
    "inProgress": false,
  },
  "appState": "STARTING",
  "auth": Object {
    "error": "",
    "inProgress": false,
    "user": null,
  },
  "flocks": Object {
    "data": Object {
      "flock1": Object {
        "name": "Flock 1",
        "owneBy": "UserId1",
      },
    },
    "error": "Error Message",
    "inProgress": false,
  },
  "initialUrl": null,
  "joinForm": Object {
    "error": "",
    "inProgress": false,
  },
  "userSettings": Object {
    "data": Object {
      "item1": "item1value",
      "item2": "item2value",
    },
    "error": "",
    "inProgress": false,
  },
}
`;

exports[`firebaseReducer reducer GET_FLOCK_REQUESTED 1`] = `
Object {
  "addForm": Object {
    "error": "",
    "inProgress": false,
  },
  "appState": "STARTING",
  "auth": Object {
    "error": "",
    "inProgress": false,
    "user": null,
  },
  "flocks": Object {
    "data": Object {},
    "error": "",
    "inProgress": true,
  },
  "initialUrl": null,
  "joinForm": Object {
    "error": "",
    "inProgress": false,
  },
  "userSettings": Object {
    "data": Object {
      "item1": "item1value",
      "item2": "item2value",
    },
    "error": "",
    "inProgress": false,
  },
}
`;

exports[`firebaseReducer reducer missing initial state 1`] = `
Object {
  "addForm": Object {
    "error": "",
    "inProgress": false,
  },
  "appState": "STARTING",
  "auth": Object {
    "error": "",
    "inProgress": false,
    "user": null,
  },
  "chickens": Object {
    "data": Object {},
    "error": "",
    "inProgress": false,
  },
  "eggs": Object {
    "data": Object {},
    "error": "",
    "inProgress": false,
  },
  "flocks": Object {
    "data": Object {},
    "error": "",
    "inProgress": false,
  },
  "initialUrl": null,
  "joinForm": Object {
    "error": "",
    "inProgress": false,
  },
  "userSettings": Object {
    "data": Object {},
    "error": "",
    "inProgress": false,
  },
}
`;

exports[`firebaseReducer reducer undefined initial state 1`] = `
Object {
  "addForm": Object {
    "error": "",
    "inProgress": false,
  },
  "appState": "STARTING",
  "auth": Object {
    "error": "",
    "inProgress": false,
    "user": null,
  },
  "chickens": Object {
    "data": Object {},
    "error": "",
    "inProgress": false,
  },
  "eggs": Object {
    "data": Object {},
    "error": "",
    "inProgress": false,
  },
  "flocks": Object {
    "data": Object {},
    "error": "",
    "inProgress": false,
  },
  "initialUrl": null,
  "joinForm": Object {
    "error": "",
    "inProgress": false,
  },
  "userSettings": Object {
    "data": Object {},
    "error": "",
    "inProgress": false,
  },
}
`;
